apps/api/src/schema/catalogue.ts

```diff
@@ -0,0 +1,91 @@
+import { z } from "@hono/zod-openapi";
+
+export const sampleProgramsQuerySchema = z.object({
+  id: z.string().optional().openapi({
+    description: "Filter sample programs by program ID and returns all programs if omitted.",
+    example: "computerscience_bs",
+  }),
+});
+
+const STANDING_YEARS = ["Freshman", "Sophomore", "Junior", "Senior"] as const;
+
+export const standingyearEnum = z.enum(STANDING_YEARS);
+
+export const courseEntrySchema = z.discriminatedUnion("type", [
+  z.object({
+    type: z.literal("courseId"),
+    value: z.string().openapi({
+      description: "A string known to be a valid course ID",
+      example: "I&CSCI31",
+    }),
+  }),
+  z.object({
+    type: z.literal("unknown"),
+    value: z.string().openapi({
+      description: "Text that could not be validated (e.g., general education requirements)",
```

Comment:
This documentation might give users the wrong idea if we attempt to parse GE requirement indicators later. In fact, those seem like somewhat low-hanging fruit, so that might be the next thing we add! I don't think the "e.g." part here is necessary, because the example probably explains it well enough.